lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -1002,6 +1030,9 @@ def __init__(
         self.relationship_name = relationship_name
         self.certificate_requests = certificate_requests
         self.mode = mode
+        self._private_key = (
+            private_key if self._validate_user_provided_private_key(private_key) else None
+        )
```

Comment:
We could probably raise earlier. If the user provided private key is not valid, no need to go any further.

Reply:
Done